Update Models methods & properties (WIP)

alanvww opened the pull request:
This PR is for updating methods and properties for each models. @ziyuan-linn Peter, thanks for the amazing works of the library and commenting everything! This update is based on the those content and also the latest examples in the collection. Feel free to review and leave any comments! 
@QuinnHe @myrahsa 

Progress
- [x] BodyPose
- [x] HandPose 
- [x] FaceMesh 
- [x] Body Segmentation
- [x] Image Classifier
- [x] Sound Classifier 🚧
- [x] Sentiment 
- [x] NN

alanvww commented:
Progress
- [x] BodyPose
- [ ] HandPose 🚧
- [ ] FaceMesh 🚧
- [ ] Body Segmentation
- [ ] Image Classifier 🚧
- [ ] Sentiment

QuinnHe commented:
Hi @alanvww , thanks for the amazing updates! One thing I noticed, can we include options that are shared by MoveNet and Blaz
<img width="960" alt="Screenshot 2024-06-17 at 8 44 38 PM" src="https://github.com/ml5js/ml5-website-v02-docsify/assets/31847981/7241efab-1252-4460-bf59-03b74f818748">
ePose. See the screenshot for reference.

alanvww commented:
@QuinnHe got it! I will work on it now :)

alanvww commented:
Quick update for the docs! Just confirmed with Peter, all `score` labels will be replaced with `confidence` in 1.0.  [Check PR here](https://github.com/ml5js/ml5-next-gen/pull/131) @QuinnHe @myrahsa @MOQN

alanvww commented:
Resolved conflicts for updated to main and also included #112

QuinnHe commented:
@alanvww some bodyseg really new PRs and Issues for your reference, the second one seems more urgent to be included: 
switch default runtime to tfjs: https://github.com/ml5js/ml5-next-gen/pull/140
BodySegmentation: Add raw values of the detected body parts to result: https://github.com/ml5js/ml5-next-gen/pull/139

alanvww commented:
> @alanvww some bodyseg really new PRs and Issues for your reference, the second one seems more urgent to be included: switch default runtime to tfjs: [ml5js/ml5-next-gen#140](https://github.com/ml5js/ml5-next-gen/pull/140) BodySegmentation: Add raw values of the detected body parts to result: [ml5js/ml5-next-gen#139](https://github.com/ml5js/ml5-next-gen/pull/139)

I can fix it quickly now. But I didn't realize NN's workload was so huge. I better skip Sound Classifier and focus on this. 

In the meantime, would you please also check the preview deployment of this PR to see the new content I added? I tried my best to match the source code, but it might still be missing something.

alanvww commented:
@QuinnHe Got it! For the runtime change, I added the `Runtime` section in the glossary update. #84 Would you please review and merge this as well? I corrected the content and runtime is the only new section there. I don't think I'll add anything else before tomorrow's launch.

QuinnHe commented:
> @QuinnHe Got it! For the runtime change, I added the `Runtime` section in the glossary update. #84 Would you please review and merge this as well? I corrected the content and runtime is the only new section there. I don't think I'll add anything else before tomorrow's launch.

Thanks Alan, will review soon!

QuinnHe commented on docs/reference/bodypose.md:
```diff
@@ -336,4 +395,39 @@ bodypose.detect(media, ?callback);
+      <img style="display:block; max-width:30%" alt="MoveNet keypoint diagram" src="https://camo.githubusercontent.com/c3641b718d7e613b2ce111a6a4575e88ca35a60cb325efdd9113c453b2a09301/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f6d6f76656e65742f636f636f2d6b6579706f696e74732d3530302e706e67">
+  </center> <br/>
+   <center>
+      <h3>BlazeNet</h3>
```
Should be "BlazePose" instead of "BlazeNet"

QuinnHe commented on docs/reference/body-segmentation.md:
```diff
@@ -158,9 +158,13 @@ const bodySegmentation = ml5.bodySegmentation(?modelName, ?options, ?callback);
 
-- **modelName**: OPTIONAL: A string specifying which model to use, "SelfieSegmentation" or "BodyPix".
+- **modelName**: Optional. A string specifying which model to use. Types of model:
+  - _BodyPix_ (default): A general-purpose model that can be used to segment people and body parts.
```
SelfieSegmentation is the default. Check https://github.com/ml5js/ml5-next-gen/blob/main/src/BodySegmentation/index.js#L31 for reference.

QuinnHe commented on docs/reference/body-segmentation.md:
```diff
@@ -170,18 +174,25 @@ const bodySegmentation = ml5.bodySegmentation(?modelName, ?options, ?callback);
   [More info on options for BodyPix model.](https://github.com/tensorflow/tfjs-models/blob/master/body-segmentation/src/body_pix/README.md#create-a-detector)
 
-- **callback(bodySegmentation, error)**: OPTIONAL. A function to run once the model has been loaded. Alternatively, call `ml5.bodySegmentation()` within the p5 `preload` function.
+- **callback(bodySegmentation)**: OPTIONAL. A function to run once the model has been loaded. Alternatively, call `ml5.bodySegmentation()` within the p5 `preload` function.
```
I remembered we still have error, it's just we don't use it in example code. We could double check this after soft launch, feel free to create a Issue for this.

alanvww replied:
Ok, I will consider putting it back with the `?` indicating that it is optional. It is actually quite important for debugging.

QuinnHe commented on docs/reference/sentiment.md:
```diff
@@ -219,32 +219,35 @@ That's it! You have successfully built a Sentiment Analysis model that predicts
-#### .predict()
+### sentiment.predict()
 
 > Given a number, will make magicSparkles
```
This does not make a lot sense to me? Maybe we can remove this? Or Alan you have any specific reasons for keeping it?

alanvww replied:
I did this to be consistent with the methods in the other models page. But I also realize that NN has a completely different structure compared to other models. Should we use the `.functionName()` style for all the pages?

QuinnHe replied:
> I did this to be consistent with the methods in the other models page. But I also realize that NN has a completely different structure compared to other models. Should we use the `.functionName()` style for all the pages?

Oh no I refer to `> Given a number, will make magicSparkles` this line, maybe we should remove this line?

alanvww replied:
Ohh got it. At least for this one, we should remove it. But the same question still, in NN a lot of `>` are working as method description. Should I reformat the whole thing now?

QuinnHe replied:
> Ohh got it. At least for this one, we should remove it. But the same question still, in NN a lot of `>` are working as method description. Should I reformat that too? I might need to work on it tomorrow morning if that is ok :/

Yes, if time is permitted, let's remove `>` for NN as well. We can do a merge tomorrow morning around 10-11am ish before the presentation. If can't make it, it's fine, we can keep `>` for now.

alanvww replied:
👌, that is feasible timing.

QuinnHe commented on docs/reference/sentiment.md:
Let's update "Properties" style to align with "Methods". We can remove `---` divider, and remove `>`.

QuinnHe commented:
@alanvww , I've done reviewing the models you have right now! I have to say, this is AWESOME! Thank you so much for your work Alan!!

alanvww commented:
> @alanvww , I've done reviewing the models you have right now! I have to say, this is AWESOME! Thank you so much for your work Alan!!

Thanks Quinn! Thanks for the reviews. My schedule is too fragmented these days and sometimes I can't finish reading the codes right away, so I have to go over things again and again. I wish I didn't mess things up too much lol.

QuinnHe commented:
Hey @alanvww , just let you know that I will start to work on SoundClassifier now, let's avoid duplicated work :)

QuinnHe commented:
> > @alanvww , I've done reviewing the models you have right now! I have to say, this is AWESOME! Thank you so much for your work Alan!!
> 
> Thanks Quinn! Thanks for the reviews. My schedule is too fragmented these days and sometimes I can't finish reading the codes right away, so I have to go over things again and again. I wish I didn't mess things up too much lol.

Haha as always you did awesome, no worries at all!

alanvww commented:
> Hey @alanvww , just let you know that I will start to work on SoundClassifier now, let's avoid duplicated work :)

All good!

QuinnHe commented on docs/reference/bodypose.md:
```diff
@@ -240,12 +240,52 @@ let bodypose = ml5.bodypose(?model, ?options, ?callback);
+    - String: The type of model to use. Default: "MULTIPOSE_LIGHTNING".
+  - _enableSmoothing_ - Optional
+    - Boolean: Whether to smooth the pose landmarks across different input images to reduce jitter. Default: true.
+
```
We may need to add `Options for MoveNet model only` as well. See https://github.com/ml5js/ml5-next-gen/blob/main/src/BodyPose/index.js#L27-L32 for reference

alanvww replied:
Thanks for point that out! It was in the previous commit but I accidentally removed it when solving conflict when updating. Will fix it now.

alanvww commented:
@QuinnHe @MOQN Last comment before bed. I just added `Properties` to the model pages. 

Question about NN: The origin methods section of it is more like an instruction. The way I modify it right now is to make it match with other page's style more. But I think those contents are still very helpful for beginners. 

Maybe we should separate it from the methods and use it in the guide?

QuinnHe commented:
> This PR is for updating methods and properties for each models. @ziyuan-linn Peter, thanks for the amazing works of the library and commenting everything! This update is based on the those content and also the latest examples in the collection. Feel free to review and leave any comments! @QuinnHe @myrahsa
> 
> Progress
> 
> * [x]  BodyPose
> * [x]  HandPose
> * [x]  FaceMesh
> * [x]  Body Segmentation
> * [x]  Image Classifier
> * [x]  Sound Classifier 🚧
> * [x]  Sentiment
> * [x]  NN

Sound Classifier is done. #[0649346](https://github.com/ml5js/ml5-website-v02-docsify/commit/0649346485ddf3f8899e8c7119b6f6a75eb65c47)

QuinnHe commented:
> @QuinnHe @MOQN Last comment before bed. I just added `Properties` to the model pages.
> 
> Question about NN: The origin methods section of it is more like an instruction. The way I modify it right now is to make it match with other page's style more. But I think those contents are still very helpful for beginners.
> 
> Maybe we should separate it from the methods and use it in the guide?

Thanks @alanvww for the work on `Properties`. After soft launch, we can consider put `How to initialize NN` uder `ml5.neuralNetwork()`.

One small thing - can we make sure the `Properties` goes before `Methods`?

alanvww commented:
> > @QuinnHe @MOQN Last comment before bed. I just added `Properties` to the model pages.
> 
> > 
> 
> > Question about NN: The origin methods section of it is more like an instruction. The way I modify it right now is to make it match with other page's style more. But I think those contents are still very helpful for beginners.
> 
> > 
> 
> > Maybe we should separate it from the methods and use it in the guide?
> 
> 
> 
> Thanks @alanvww for the work on `Properties`. After soft launch, we can consider put `How to initialize NN` uder `ml5.neuralNetwork()`.
> 
> 
> 
> One small thing - can we make sure the `Properties` goes before `Methods`?
> 
> 
> 
> 

Got it. I am on my way to school now. Will fix that soon.

alanvww commented:
@QuinnHe this PR is ready for merge now!

QuinnHe commented:
Thanks @alanvww ! This looks great!
